src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -3068,6 +3068,118 @@ public override MemberInfo[] GetMember(string name, MemberTypes type, BindingFla
 
             return compressMembers;
         }
+
+        public override MemberInfo? GetMemberWithSameMetadataDefinitionAs(MemberInfo member)
```

Comment:
Do we need to override this on any other Type derivations, e.g. any of the reflection emit types, anything from MetadataLoadContext, etc.?

Reply:
You can also provide fallback implementation here that enumerates all items using public API and finds the matching one using `HasSameMetadataDefinitionAs`.

Reply:
Thanks for the suggestion, @jkotas. I've implemented it.